### PR TITLE
[react-native-paypal-lib]: Adds new lib types

### PR DESCRIPTION
**Pull request by gstvds**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by peterblazejewicz — COMMENTED** [review]
@gstvds thx for the PR!
It seems the package ships its own definition files:
https://unpkg.com/browse/react-native-paypal-lib@1.0.9/index.d.ts
What's the purpose of trying to add this definition?

**Comment by gstvds** [comment]
@peterblazejewicz 
This package was last updated about 2 years ago and it seems the owner is inactive too. The declaration is wrong (package name is `react-native-paypal-lib` and the declaration is appointing to `react-native-paypal` which is [another package](https://github.com/smarkets/react-native-paypal#readme)) so TS don't recognize.

File: types/react-native-paypal-lib/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    description: string;
+    acceptCreditCards: boolean;
+}
+
+// tslint:disable-next-line no-unnecessary-class
+export default class RNPaypal {
+    static paymentRequest(params: PaymentParams): Promise<PaymentConfirmation>;
+    static ENVIRONMENT: EnvironmentParam;
+    static INTENT: IntentParam;
+}
```
Comment (peterblazejewicz) [inline types/react-native-paypal-lib/index.d.ts:84]:
```suggestion
declare const RNPaypal: {
    paymentRequest(params: PaymentParams): Promise<PaymentConfirmation>;
    readonly ENVIRONMENT: EnvironmentParam;
    readonly INTENT: IntentParam;
};

export default RNPaypal;
```
just proposition of another approach (suggested in the no-unnecessary-class details)

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@gstvds thx for the contribution

note for maintainer (second pari of 👀 )
please review question about the license and purpose of this package.

**Review by peterblazejewicz — APPROVED** [review]

**Comment by orta** [comment]
Cool, yeah, that explanation makes sense to me - thanks folks

**Pull request merged**